wal/decoder.go

```diff
@@ -104,7 +107,7 @@ func (d *decoder) decodeRecord(rec *walpb.Record) error {
 		}
 	}
 	// record decoded as valid; point last valid offset to end of record
-	d.lastValidOff += recBytes + padBytes + 8
+	d.lastValidOff += recBytes + padBytes + frameSizeBytes
```

Comment:
since going through the trouble of changing this, have:
```go
d.lastValidOff += frameSizeBytes + recBytes + padBytes
```
to correspond with the physical layout of the frame that's being added